File: src/main/java/com/multicampus/gamesungcoding/a11ymarketserver/feature/cart/repository/CartRepository.java
package com.multicampus.gamesungcoding.a11ymarketserver.feature.cart.repository;

import com.multicampus.gamesungcoding.a11ymarketserver.feature.cart.entity.Cart;
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.Users;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.UUID;

public interface CartRepository extends JpaRepository<Cart, UUID> {
    Cart findByUser(Users user);

    Cart findByUser_UserId(UUID userId);
}

File: src/main/kotlin/com/multicampus/gamesungcoding/a11ymarketserver/feature/cart/repository/CartRepository.kt
package com.multicampus.gamesungcoding.a11ymarketserver.feature.cart.repository

import com.multicampus.gamesungcoding.a11ymarketserver.feature.cart.entity.Cart
import com.multicampus.gamesungcoding.a11ymarketserver.feature.user.entity.Users
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface CartRepository : JpaRepository<Cart, UUID> {
    fun findByUser(user: Users): Cart?

    fun findByUserUserId(userId: UUID): Cart?
}
